includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -702,10 +702,14 @@ public function prepare_order_source( $order = null ) {
 				$source_object = WC_Stripe_API::retrieve( 'sources/' . $source_id );
 			} elseif ( apply_filters( 'wc_stripe_use_default_customer_source', true ) ) {
 				/*
-				 * We can attempt to charge the customer's default source
-				 * by sending empty source id.
+				 * ~We can attempt to charge the customer's default source~
+				 * ~by sending empty source id.~
+				 * We retrieve the default customer source manually.
 				 */
-				$stripe_source = '';
+				$default_source = $this->get_default_customer_source_for_order( $order );
```

Comment:
At this point, we're trying to get a source for the renewal order of the format:

```
(object) array(
	'token_id'      => '',
	'customer'      => $stripe_customer->get_id(),
	'source'        => $source_object->id,
	'source_object' => $source_object,
)
```

We're retrieving the default source, but why do we need to do that? Can't we rely on making the request to Stripe with just the `customer` information, and no `source` information?

When the transaction goes through, we could save the source to the order using something like:

```
$this->save_source_to_order( $renewal_order, (object) array(
	'customer' => $response->customer,
	'source'      => $response->source,
));
```

at this line: https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1040/files#diff-0e15c5ae3c4b28bb406663efdd18e1d2R290

Reply:
> We're retrieving the default source, but why do we need to do that? Can't we rely on making the request to Stripe with just the customer information, and no source information?

I think my response in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1040#discussion_r344757880 covers that.

> When the transaction goes through, we could save the source to the order using something like:

AFAICT the values generated by `prepare_order_source` should already be the same. Is there a reason you think the format you suggested is better than what's already in there? Maybe I'm missing some scenario.